elyra/airflow/bootstrapper.py

```diff
@@ -74,10 +75,28 @@ def __init__(self, **kwargs: Any) -> None:
         # Infer secure from the endpoint's scheme.
         self.secure = self.cos_endpoint.scheme == 'https'
 
-        self.cos_client = minio.Minio(self.cos_endpoint.netloc,
-                                      access_key=os.getenv('AWS_ACCESS_KEY_ID'),
-                                      secret_key=os.getenv('AWS_SECRET_ACCESS_KEY'),
-                                      secure=self.secure)
+        # get minio credentials provider
+        if "cos-user" in self.input_params and "cos-password" in self.input_params:
+            cred_provider = providers.StaticProvider(
+                access_key=self.input_params.get("cos-user"),
+                secret_key=self.input_params.get("cos-password"),
+            )
+        elif "AWS_ACCESS_KEY_ID" in os.environ and "AWS_SECRET_ACCESS_KEY" in os.environ:
+            cred_provider = providers.EnvAWSProvider()
+        elif "AWS_ROLE_ARN" in os.environ and "AWS_WEB_IDENTITY_TOKEN_FILE" in os.environ:
```

Comment:
How is the role being propagated to the container? Are you assuming that the container would have the info and use the same for all users?

Reply:
This is how IRSA works, it injects `AWS_ROLE_ARN` and  `AWS_WEB_IDENTITY_TOKEN_FILE` based of the Kubernetes Pod's ServiceAccount.

If the user has chosen to not pass `cos_user` and `cos_password` we check if they have set up IRSA (out of band, with respect to Elyra).

Does that answer your question?